mlx/ops.h

```diff
@@ -668,6 +668,30 @@ array sign(const array& a, StreamOrDevice s = {});
 /** Logical not of an array */
 array logical_not(const array& a, StreamOrDevice s = {});
 
+/** Logical and of two arrays */
+array logical_and(const array& a, const array& b, StreamOrDevice s = {});
+array operator&&(const array& a, const array& b);
+template <typename T>
+array operator&&(T a, const array& b) {
+  return logical_and(array(a), b);
+}
+template <typename T>
+array operator&&(const array& a, T b) {
+  return logical_and(a, array(b));
+}
```

Comment:
If `operator&&` is only supported for bools (as I think it should be). I think you can just specialize these for `bool`, don't need templates here. Right?